modules/services/window-managers/spectrwm.nix

```diff
@@ -0,0 +1,142 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.xsession.windowManager.spectrwm;
+
+  renderedValue = val:
+    if isBool val then if val then "1" else "0" else toString val;
+
+  renderedSettings = settings:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "${name} = ${renderedValue value}") settings);
+
+  renderedBindings = bindings:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "bind[${name}] = ${value}") bindings);
+
+  renderedUnbindings = unbindings:
+    concatStringsSep "\n" (map (value: "bind[] = ${value}") unbindings);
+
+  renderedPrograms = programs:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "program[${name}] = ${value}") programs);
+
+  renderedQuirks = quirks:
+    concatStringsSep "\n"
+    (mapAttrsToList (name: value: "quirk[${name}] = ${value}") quirks);
+
+  settingType = types.oneOf [ types.bool types.int types.str ];
+
+in {
+  meta.maintainers = [ hm.maintainers.loicreynier ];
+
+  options = {
+    xsession.windowManager.spectrwm = {
+      enable = mkEnableOption "Spectrwm window manager";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.spectrwm;
+        defaultText = literalExpression "pkgs.spectrwm";
+        description = ''
+          Package providing the <command>spectrwm</command> command.
+        '';
+      };
+
+      settings = mkOption {
+        type = types.attrsOf settingType;
+        default = { };
+        example = literalExpression ''
+          {
+            modkey = "Mod4";
+            workspace_limit = 5;
+            focus_mode = "manual";
+            focus_close = "next";
+          }
+        '';
+        description = "Spectrwm settings.";
+      };
+
+      bindings = mkOption {
+        type = types.attrsOf types.string;
+        default = { };
+        example = literalExpression ''
+          {
+            term = "Mod+Return";
+            restart = "Mod+Shift+r";
+            quit = "Mod+Shift+q";
+          }
+        '';
```

Comment:
Can you use a command as a key? Like "alacritty --some-flag"?

If so, the `programs` option is redundant when you can manage everything with nix expressions (IMO).

Reply:
The `action` key such as the `program` identifer must be a single word.